test/fw/bin/pbs_snapshot

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (C) 1994-2016 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+import sys
+import os
+import getopt
+import tempfile
+import logging
+import logging.config
+import time
+import errno
+import signal
+
+import ptl
+from ptl.lib.pbs_testlib import *
+from ptl.utils.pbs_snaputils import *
+from ptl.utils.pbs_dshutils import DshUtils
+
+
+# trap SIGINT and SIGPIPE
+def trap_exceptions(etype, value, tb):
+    sys.excepthook = sys.__excepthook__
+    if issubclass(etype, KeyboardInterrupt):
+        pass
+    elif issubclass(etype, IOError) and value.errno == errno.EPIPE:
+        pass
+    else:
+        sys.__excepthook__(etype, value, tb)
+sys.excepthook = trap_exceptions
+
+
+def usage():
+    msg = """
+Usage: pbs_snapshot -o <path to output tar file> [OPTION]
+
+    Take snapshot of a PBS system and optionally capture logs for diagnostics
+
+    -d <pbs_diag>                     diag directory to use as input
+    -H <hostname>                     hostname to operate on
+                                      Defaults to value of PBS_SERVER
+    -l <loglevel>                     set log level to one of INFO, INFOCLI,
+                                      INFOCLI2, DEBUG, DEBUG2, WARNING, ERROR
+                                      or FATAL
+    --service-logs=<num days>         number of service logs to collect
```

Comment:
This came up during the EDD discussion, I think @billnitzberg had mentioned it first, and I thought it made sense because, technically, even accounting logs are generated by a daemon (the server), so "service logs" sounds more technically correct in the context of differentiating accounting logs with others. To be honest, I wasn't sure whether the term "service logs" was popular or not, if people won't be able to associate them with the server, scheduler, mom and comm logs, then we should probably change the name. What do you guys think?